### ISSUE TITLE
Qed fails to check a term produced by the destruct tactic

### ISSUE THREAD

**Issue by rafoo** [coq/coq#10364]
#### Description of the problem

Consider the following snippet:

```
Inductive isId {A : Set} : forall {B : Set}, (A -> B) -> Set := isId_id : isId (@id A).

Lemma should_not_be_provable (A : Set) (f : A -> A) : isId f -> f = @id A.
Proof.
  intro H.
  destruct H.
  reflexivity.
Qed.
```

On this snippet, the refiner let me complete the proof (it shows `No more subgoals.` after `reflexivity`) but `Qed` fails with the following error: 

```
Error: Illegal application: 
The term "@eq" of type "forall A : Type, A -> A -> Prop"
cannot be applied to the terms
 "B -> B" : "Set"
 "a" : "A -> B"
 "id (A:=B)" : "B -> B"
The 2nd term has type "A -> B" which should be coercible to "B -> B".
```

I do not expect this lemma to be provable (because I think that it is not compatible with univalence) so I think that the `destruct` tactic should fail in a similar way to the following failed proof attempt:

```
Lemma is_not_provable (A : Set) (a : A) (H : a = a) : H = eq_refl.
Proof.
  destruct H.
```

```
Error: Cannot instantiate metavariable P of type "forall a0 : A, a = a0 -> Prop" with abstraction
"fun (a : A) (H : a = a) => H = eq_refl" of incompatible type "forall a : A, a = a -> Prop".
```

#### Coq Version

```
$ coqtop -v
The Coq Proof Assistant, version 8.9.1 (June 2019)
compiled on Jun 11 2019 18:20:55 with OCaml 4.07.1
```

**Comment by ejgallego** [comment]
Fixed in 8.10.

**Comment by rafoo** [comment]
Thank you @ejgallego and sorry for the noise!
